kong/pdk/client.lua

```diff
@@ -148,6 +148,50 @@ local function new(self)
   end
 
 
+  ---
+  -- Returns the consumer from the datastore (or cache).
+  -- Will look up the consumer by id, and optionally will do a second search by name.
+  -- @phases access, header_filter, body_filter, log
+  -- @tparam consumer_id string. The consumer id to look up.
+  -- @tparam [opt] search_by_username boolean. If truthy,
+  -- then if the consumer was not found by id,
+  -- then a second search by username will be performed
+  -- @treturn table|nil consumer entity or nil
+  -- @treturn nil|err nil if success, or error message if failure
+  -- @usage
+  -- local consumer_id = "john_doe"
+  -- local consumer = kong.client.load_consumer(consumer_id, true)
+  function _CLIENT.load_consumer(consumer_id, search_by_username)
+    check_phase(AUTH_AND_LATER)
+
+    if not consumer_id or type(consumer_id) ~= "string" then
+      error("consumer_id must be a string", 2)
+    end
+
+    if not utils.is_valid_uuid(consumer_id) and not search_by_username then
```

Comment:
Minor comment here, not worthy of not approving: `utils.is_valid_uuid(consumer_id)` is called twice on this function. It would be nice it that happened only once. I realize that this particular `if` here is ... delicate. So perhaps just set a local variable `local is_valid_uuid = utils.is_valid_uuid(consumer_id)` at the beginning?